apps/server/src/modules/common-cartridge/export/utils.ts

```diff
@@ -0,0 +1,45 @@
+import { InternalServerErrorException } from '@nestjs/common';
+import { ObjectID } from 'bson';
+import { Builder } from 'xml2js';
+
+export type OmitVersion<T> = Omit<T, 'version'>;
+
+export type OmitVersionAndFolder<T> = Omit<T, 'version' | 'folder'>;
+
+export type OmitVersionAndType<T> = Omit<T, 'version' | 'type'>;
+
+const xmlBuilder = new Builder({
+	xmldec: { version: '1.0', encoding: 'UTF-8' },
+	renderOpts: { pretty: true, indent: '    ', newline: '\n' },
+});
+
+export function buildXmlString(obj: unknown): string {
+	return xmlBuilder.buildObject(obj);
+}
+
+export function createVersionNotSupportedError(version: string): Error {
+	return new InternalServerErrorException(`Common Cartridge version ${version} is not supported`);
```

Comment:
Maybe not this place if the error catch again, but we always want to use logger instead of legacyLogger and Loggable / LoggableExceptions. You must not write test for the loggables, but it should be use also in the commonn-cartridge module.

Edit: The wrapper around the error sound all like it should be loggableExceptions. Look into the code for examples.